### PR TITLE
Added an iOS and Android bridge to the BatchPush.getLastKnownPushToke…

### PR DESCRIPTION
**Pull request by tootsweet**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,16 +1,16 @@
 {
-  "name": "react-native-batch-push",
```
Comment (Minishlink) [inline package.json:2]: Can you revert this?

File: android/src/main/java/tech/bam/RNBatchPush/RNBatchPushModule.java [hunk 2]
```diff
@@ -54,6 +54,16 @@ private void startBatch() {
 
     Batch.onStart(activity);
   }
+  
+  @ReactMethod
+  public void lastKnownPushToken(Promise promise) {
+    try{
```
Comment (Minishlink) [inline android/src/main/java/tech/bam/RNBatchPush/RNBatchPushModule.java:60]: Why catching an error on Android but not on iOS?

File: README.md [hunk 3]
```diff
@@ -142,3 +142,11 @@ BatchPush.fetchNewNotifications('theUserId', 'authKey')
   })
   .catch(e => console.warn('BatchPush error', e));
 ```
+
+### Last known push token
+
```
Comment (Minishlink) [inline README.md:147]: Please add a warning that this should only be used for debugging - the correct way in production would be to [use a listener](https://batch.com/doc/android/advanced/getting-registration.html#_listening-to-the-broadcast)

**Review by Minishlink — COMMENTED** [review]
Thanks for your contribution! :) Can you review my comments?

File: package.json (hunk 1 above)
Comment (tootsweet) [reply package.json:2]: Sure

File: README.md (hunk 3 above)
Comment (tootsweet) [reply README.md:147]: It is my understanding that Batch already implements a listener and stores the push token when it changes. Why add one more listener on top of that?

Comment (Minishlink) [reply README.md:147]: I'm only referencing what states in the doc :) It looks like we shouldn't rely too much on this value.

> This method will return null if no registration identifier has been fetched or if Batch isn't started, meaning that you have to call this method after Batch.onStart(). It is useful if you want to show the token in your UI, for example in a debug view.

Comment (abarisain) [reply README.md:147]: This (annoying) limitation is exactly why we introduced the listener down the road: users had to work around the token not being available with something like setTimeout.

The listener also gives you the token provider (GCM/FCM, and possibly more in the future)

My 2cts is that both can be used: they're not meant for the same use cases.

**Pull request closed without merging**